src/server/controller/cycleData/persistNodeValue/expressionEvaluator/functions/validatorSubCategory.ts

```diff
@@ -15,7 +15,7 @@ export const validatorSubCategory: ExpressionFunction<Context> = {
         Objects.isEmpty(categoryValue) ||
         Numbers.greaterThanWithTolerance(
           categoryValue,
-          Numbers.sum(subCategoryValues?.filter((v) => !Objects.isEmpty(v)))
+          Numbers.sum(subCategoryValues?.filter((v) => !Objects.isEmpty(v))) ?? 0
```

Comment:
`Numbers.sum` : returns null if array empty
`Numbers.greaterThanWithTolerance(` : Returns false if compared nullish value

Reply:
The problem is in the logic.
Before greaterThan ensure arguments are not empty. If one of them is empty , then validation is valid .

Leave sum as it was . 

Does it work ?